ArduCopter/mode_auto.cpp

```diff
@@ -717,7 +717,7 @@ Return true if we do not recognize the command so that we move on to the next co
 //      we double check that the flight mode is AUTO to avoid the possibility of ap-mission triggering actions while we're not in AUTO mode
 bool ModeAuto::verify_command(const AP_Mission::Mission_Command& cmd)
 {
-    if (copter.flightmode != &copter.mode_auto) {
+    if (copter.flightmode->mode_number() != Mode::Number::AUTO) {
```

Comment:
This breaks the new auto RTL mode. It is really auto underneath but will report as a different number. You could do:

 ```suggestion
    if ((copter.flightmode->mode_number() != Mode::Number::AUTO)  && (copter.flightmode->mode_number() != Mode::Number::AUTO_RTL)) {
```

but I'm not sure that is better than the current method.

Reply:
@IamPete1 san.
The pointer for the current flight mode is set to copter .flightmode.
Therefore, we recognize that there cannot be two flight modes with the same pointer.
Am I wrong in my perception?